// nextra-theme-docs/dist/locale-switch.js
function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var React = _interopDefault(require('react'));
var router = require('next/router');
var Link = _interopDefault(require('next/link'));
var cn = _interopDefault(require('classnames'));

function LocaleSwitch({
  options
}) {
  const {
    locale,
    asPath
  } = router.useRouter();
  return /*#__PURE__*/React.createElement("details", {
    className: "locale-switch relative"
  }, /*#__PURE__*/React.createElement("summary", {
    className: "text-current p-2 cursor-pointer outline-none"
  }, /*#__PURE__*/React.createElement("svg", {
    fill: "none",
    viewBox: "0 0 24 24",
    width: "24",
    height: "24",
    stroke: "currentColor"
  }, /*#__PURE__*/React.createElement("path", {
    strokeLinecap: "round",
    strokeLinejoin: "round",
    strokeWidth: 2,
    d: "M3 5h12M9 3v2m1.048 9.5A18.022 18.022 0 016.412 9m6.088 9h7M11 21l5-10 5 10M12.751 5C11.783 10.77 8.07 15.61 3 18.129"
  }))), /*#__PURE__*/React.createElement("ul", {
    className: "locale-dropdown absolute right-0 block bg-white dark:bg-dark dark:border-gray-700 py-1 rounded border"
  }, options.map(l => /*#__PURE__*/React.createElement(Link, {
    key: l.locale,
    href: asPath,
    locale: l.locale
  }, /*#__PURE__*/React.createElement("a", {
    className: cn("block no-underline text-current py-2 px-4 hover:bg-gray-200 dark:hover:bg-gray-800 whitespace-nowrap", {
      'font-semibold': locale === l.locale,
      'bg-gray-100 dark:bg-gray-900': locale === l.locale
    })
  }, l.text)))));
}

module.exports = LocaleSwitch;
